test/metadata/main.go

```diff
@@ -0,0 +1,306 @@
+		return
+	}
+
+	providerApp := credsMap["providerApp"].(map[string]interface{})
+	clientId := providerApp["clientId"].(string)
+	clientSecret := providerApp["clientSecret"].(string)
+	accessToken := credsMap["AccessToken"].(map[string]interface{})["Token"].(string)
+	refreshToken := credsMap["RefreshToken"].(map[string]interface{})["Token"].(string)
+
+	salesforceSubdomain := credsMap["providerWorkspaceRef"].(string)
+
+	cfg := &oauth2.Config{
+		ClientID:     clientId,
+		ClientSecret: clientSecret,
+		Endpoint: oauth2.Endpoint{
+			AuthURL:   fmt.Sprintf("https://%s.my.salesforce.com/services/oauth2/authorize", salesforceSubdomain),
+			TokenURL:  fmt.Sprintf("https://%s.my.salesforce.com/services/oauth2/token", salesforceSubdomain),
+			AuthStyle: oauth2.AuthStyleInParams,
+		},
+	}
+
+	tok := &oauth2.Token{
+		AccessToken:  accessToken,
+		RefreshToken: refreshToken,
+		TokenType:    "bearer",
+		Expiry:       time.Now().Add(-1 * time.Hour),
+	}
+
+	ctx := context.Background()
+
+	sfc, err := connectors.Salesforce(
+		salesforce.WithClient(ctx, http.DefaultClient, cfg, tok,
+			salesforce.GetTokenUpdater(tok), // this is necessary to update token
+		),
+		salesforce.WithSubdomain(salesforceSubdomain),
+	)
+	if err != nil {
+		slog.Error("Error creating Salesforce connector", "error", err)
+
+		return
+	}
+
+	defer func() {
+		_ = sfc.Close()
+	}()
+
+	createObjectData, err := os.ReadFile("./metadata/testCreateCustomObject.json")
+	if err != nil {
+		slog.Error("Error opening testOperation.json", "error", err)
+		os.Exit(1)
+	}
+
+	if string(createObjectData) == "" {
+		slog.Error("Error opening testOperation.json", "error", err)
+		os.Exit(1)
+	}
+
+	// fmt.Println(string(createObjectData))
+
+	var objectOperation *common.XMLData
+
+	if err := json.Unmarshal(createObjectData, &objectOperation); err != nil {
+		slog.Error("Error marshalling testOperation.json", "error", err)
+		os.Exit(1)
+	}
+
+	// fmt.Println(operation.ToXML())
+
+	// res, err := sfc.CreateMetadata(context.Background(), objectOperation, tok.AccessToken)
+	// if err != nil {
+	// 	slog.Debug("err", "err", err)
+	// }
+
+	// fmt.Println("Object Operation Result: ", res)
+
+	createFieldData, err := os.ReadFile("./metadata/testCreateCustomField.json")
+	if err != nil {
+		slog.Error("Error opening testOperation.json", "error", err)
+		os.Exit(1)
+	}
+
+	if string(createFieldData) == "" {
+		slog.Error("Error opening testOperation.json", "error", err)
+		os.Exit(1)
+	}
+
+	// fmt.Println(string(createFieldData))
+
+	var fieldOperation *common.XMLData
+
+	if err := json.Unmarshal(createFieldData, &fieldOperation); err != nil {
+		slog.Error("Error marshalling testOperation.json", "error", err)
+		os.Exit(1)
+	}
+
+	operation := &common.XMLData{
+		XMLName:     "createMetadata",
+		Children:    []common.XMLSchema{objectOperation, fieldOperation},
+		SelfClosing: false,
+	}
+
+	res2, err := sfc.CreateMetadata(context.Background(), operation, tok)
```

Comment:
what's the difference between res1 and res2? did you forget to push a commit? this file looks unfinished.